### DISCUSSION TITLE
Welcome to mlx-graphs Discussions!

### DISCUSSION THREAD

**Discussion by TristanBilot** [mlx-graphs/mlx-graphs#106] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with mlx-graphs.

**Comment by Kh4L** [comment]
Hi @TristanBilot,
Just came across this framework and I like the project!

Is it still maintained? Any plans for further development or potential integration into another framework?

**Reply by TristanBilot** [reply]
Hi @Kh4L, thanks for your interest in the project!

Unfortunately, I didn't have time to allocate to maintain the project recently but I plan to do so in the next week or so. 

On the roadmap, an efficient method for neighbor sampling is still missing, see this (long) PR:
https://github.com/mlx-graphs/mlx-graphs/pull/112
Since we started this project, MLX has evolved and new primitives have been added so we may find a way that doesn't require specific Metal kernels, but we have to check. 

Doing more recent benchmarks with other frameworks like PyG or DGL could be useful too (see example here https://mlx-graphs.github.io/mlx-graphs/tutorials/notebooks/benchmark_pyg_dgl_mxg.html). 

We also miss many GNN layers to integrate in the framework, notably the GraphTransformer variants that could be nice to add. 

Is there any other feature you would like to integrate?

**Reply by Kh4L** [reply]
That's a great news!

> but I plan to do so in the next week or so.

Curious to see the roadmap, if you find the time 🙂  

I would be happy to help with GraphTransformer if I have the bandwidth.

> Is there any other feature you would like to integrate?

Nothing specific at the moment, but I'll let you know.
